### PR TITLE
Removes ensureYellow() after index creation in the integration tests

### PR DESCRIPTION
**Pull request by abeyad**
Removes ensureYellow() calls after index creation in the integration
tests, as they are no longer needed with index creation now waiting 
for shards to be started before returning from the index creation call 
(by default, it waits for the primary of each shard to be started 
before returning, which is what ensureYellow() was ensuring anyway).

Relates #19450

### CODE REVIEW

File: core/src/test/java/org/elasticsearch/consistencylevel/WriteConsistencyLevelIT.java [hunk 1]
```diff
@@ -30,19 +31,20 @@
...
 
 /**
  *
  */
 public class WriteConsistencyLevelIT extends ESIntegTestCase {
     public void testWriteConsistencyLevelReplication2() throws Exception {
-        prepareCreate("test", 1, Settings.builder().put("index.number_of_shards", 1).put("index.number_of_replicas", 2)).execute().actionGet();
+        CreateIndexResponse createIndexResponse =
+            prepareCreate("test", 1, Settings.builder().put("index.number_of_shards", 1).put("index.number_of_replicas", 2))
+                .execute()
```
Comment (nik9000) [inline core/src/test/java/org/elasticsearch/consistencylevel/WriteConsistencyLevelIT.java]: I'd replace this with `get()`.

File: core/src/test/java/org/elasticsearch/indexlifecycle/IndexLifecycleActionIT.java [hunk 2]
```diff
@@ -89,7 +83,8 @@ public void testIndexLifecycleActionsWith11Shards1Backup() throws Exception {
 
         // first wait for 2 nodes in the cluster
         logger.info("Running Cluster Health");
```
Comment (nik9000) [inline core/src/test/java/org/elasticsearch/indexlifecycle/IndexLifecycleActionIT.java]: Maybe change this to "waiting for replicas to be assigned" or something.

File: core/src/test/java/org/elasticsearch/indices/memory/breaker/RandomExceptionCircuitBreakerIT.java [hunk 3]
```diff
@@ -114,13 +112,11 @@ public void testBreakerWithRandomExceptions() throws IOException, InterruptedExc
...
                 .setSettings(settings)
                 .addMapping("type", mapping).execute().actionGet();
-        ClusterHealthResponse clusterHealthResponse = client().admin().cluster()
-                .health(Requests.clusterHealthRequest().waitForYellowStatus().timeout(TimeValue.timeValueSeconds(5))).get(); // it's OK to timeout here
         final int numDocs;
-        if (clusterHealthResponse.isTimedOut()) {
+        if (response.isShardsAcked() == false) {
             /* some seeds just won't let you create the index at all and we enter a ping-pong mode
              * trying one node after another etc. that is ok but we need to make sure we don't wait
              * forever when indexing documents so we set numDocs = 1 and expect all shards to fail
```
Comment (nik9000) [inline core/src/test/java/org/elasticsearch/indices/memory/breaker/RandomExceptionCircuitBreakerIT.java]: The log statement below this comment is now inaccurate.

**Comment by nik9000** [comment]
Left some minor stuff. Otherwise LGTM.

**Comment by abeyad** [comment]
@nik9000 Thanks for the review!  I'll make the changes and merge.

**Pull request merged**